File: P1.3/P1.3_obligatorio/src/RepercutirDemandaFilter.cpp
#include "RepercutirDemandaFilter.h"

void RepercutirDemandaFilter::execute(Target& t) {
    string a;

    if(t.getEstado() == EN_OFERTA) {

        double ale = Filter::aleatorio(5, 20);
        a = "La demanda del Ticker sube un " + to_string(ale) + "%";
        t.setDemanda( t.getDemanda() + ale );

    } else if ( t.getEstado() == SIN_OFERTA ) {

        double ale = Filter::aleatorio(5, 10);
        a = "La demanda del Ticker baja un " + to_string(ale) + "%";
        t.setDemanda( t.getDemanda() - ale/100 );

    } else {
        a = "Demanda neutra, no se aplica filtro";
    }

    t.addLog(a);
}

File: P1.3/P1.3_opcional/RepercutirDemandaFilter.dart
import 'Aleatorio.dart';
import 'Estado.dart';
import 'Filter.dart';
import 'Target.dart';

class RepercutirDemandaFilter extends Filter with Aleatorio {
  execute(Target t) {
    String a;

    if(t.estado == Estado.EN_OFERTA) {

        double ale = aleatorio(5, 20);
        a = "La demanda del Ticker sube un "+ ale.toStringAsFixed(4) +" %";
        t.demanda = t.demanda + ale ;

    } else if ( t.estado == Estado.SIN_OFERTA ) {

        double ale = aleatorio(5, 10);
        a = "La demanda del Ticker baja un "+ ale.toStringAsFixed(4) +" %";
        t.demanda = t.demanda - ale/100 ;

    } else {
        a = "Demanda neutra, no se aplica filtro";
    }

    t.addLog(a);
  }
}